projects/plugins/jetpack/extensions/blocks/top-posts/top-posts.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Top Posts Block.
+ *
+ * @since $$next-version$$
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Top_Posts;
+
+use Automattic\Jetpack\Blocks;
+use Jetpack_Gutenberg;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	Blocks::jetpack_register_block(
+		__DIR__,
+		array( 'render_callback' => __NAMESPACE__ . '\load_assets' )
+	);
```

Comment:
```suggestion
        if ( ! defined( 'IS_WPCOM' ) && ( ( new Connection_Manager( 'jetpack' ) )->has_connected_owner() && ! ( new Status() )->is_offline_mode() ) )
	Blocks::jetpack_register_block(
		__DIR__,
		array( 'render_callback' => __NAMESPACE__ . '\load_assets' )
	);
```

We'll need the following classes:

```PHP
use Automattic\Jetpack\Connection\Manager as Connection_Manager;
use Automattic\Jetpack\Status;
```

Reply:
Checks added - thank you!